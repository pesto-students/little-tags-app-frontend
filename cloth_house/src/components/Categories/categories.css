.categories{
    padding-top: 0.5%;
    padding-left: 2%;
    padding-right: 2%;
    padding-bottom: 0.5%;
    background-color: white;
}

.categories .category{
    color: #47525E;
    font-family: Lato;
    font-size: 15px;
    line-height: 33px;
    text-align: left;
}