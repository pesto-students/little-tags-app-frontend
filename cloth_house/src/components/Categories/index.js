import { React } from "react";
import { Row, Col,  Space, Menu, Dropdown,Divider } from "antd";
import { DownOutlined } from "@ant-design/icons";
import "./categories.css";
import {Button} from "antd"
const menu = (
  <Menu>
    <Menu.Item>
      <Button type="link">1st menu item</Button>
    </Menu.Item>
    <Menu.Item>
      <Button type="link">2nd menu item</Button>
    </Menu.Item>
    <Menu.Item>
      <Button type="link">3rd menu item</Button>
    </Menu.Item>
  </Menu>
);

const Category = (args) => {
  return (
    <>
      <Space>
        <img
          src={args.imagesrc}
          alt="fallback"
          style={{ width: "55px", height: "55px" }}
        ></img>
        <Dropdown overlay={menu} placement="bottomCenter" arrow>
          <Button type="link"
            className="ant-dropdown-link category"
            href={args.hreflink}
          >
            {args.menuLabel} <DownOutlined />
          </Button>
        </Dropdown>
      </Space>
    </>
  );
};

const Categories = (props) => {
 const redirectToParticularCategory=(category)=>{
  props.history&&props.history.push(category)
 }
  return (
    <div className="categories">
      <Row gutter={{ xs: 8, sm: 16, md: 24, lg: 32 }} align="middle" justify="left">
        <Col onClick={()=>redirectToParticularCategory("/category/clothing")} className="gutter-row " xs={16} sm={6} md={5} lg={3} xl={3}>
          <Category

            imagesrc="./assets/images/clothing.png"
            menuLabel={props.intl&&props.intl.formatMessage({id:"app.components.LangSwitch.category.clothing"})}
            //hreflink="/category/clothing"
            
          />
        </Col>
        <Divider type="vertical" style={{margin:"2px",height:"50px",borderWidth:"2px",borderColor:"#baafaf"}} />
        <Col onClick={()=>redirectToParticularCategory("/category/footwear")} className="gutter-row " xs={16} sm={6} md={5} lg={3} xl={3}>
          <Category
            imagesrc="./assets/images/footwear.jpg"
            menuLabel={props.intl&&props.intl.formatMessage({id:"app.components.LangSwitch.category.footwear"})}
           // hreflink="/category/footwear"
           
          />
        </Col>
        <Divider type="vertical" style={{margin:"2px",height:"50px",borderWidth:"2px",borderColor:"#baafaf"}}  />
        <Col onClick={()=>redirectToParticularCategory("/category/bags")} className="gutter-row " xs={16} sm={6} md={5} lg={3} xl={3}>
          <Category imagesrc="./assets/images/bags.png" 
          menuLabel={props.intl&&props.intl.formatMessage({id:"app.components.LangSwitch.category.bags"})}
          //hreflink="/category/bags"
          
          />
        </Col>
      </Row>
    </div>
  );
};

export default Categories;
