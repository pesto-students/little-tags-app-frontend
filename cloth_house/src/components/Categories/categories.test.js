import createComponentWithIntl from '../../utils/helper_snapshot'
import Categories from "./index"

it('Categories Snapshot', () => {
    const component = createComponentWithIntl(<Categories />);
    let tree = component.toJSON()
  
    expect(tree).toMatchSnapshot()
  
  });