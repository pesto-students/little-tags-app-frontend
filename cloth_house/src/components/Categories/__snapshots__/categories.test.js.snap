// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Snapshot 1`] = `
<div
  className="categories"
>
  <div
    className="ant-row ant-row-left ant-row-middle"
    style={
      Object {
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
      }
    }
  >
    <div
      className="ant-col gutter-row  ant-col-xs-16 ant-col-sm-6 ant-col-md-5 ant-col-lg-3 ant-col-xl-3"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <img
            alt="fallback"
            src="./assets/images/clothing.png"
            style={
              Object {
                "height": "55px",
                "width": "55px",
              }
            }
          />
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <button
            className="ant-btn ant-btn-link ant-dropdown-trigger ant-dropdown-link category"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            <span>
               
            </span>
            <span
              aria-label="down"
              className="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="ant-divider ant-divider-vertical"
      role="separator"
      style={
        Object {
          "borderColor": "#baafaf",
          "borderWidth": "2px",
          "height": "50px",
          "margin": "2px",
        }
      }
    />
    <div
      className="ant-col gutter-row  ant-col-xs-16 ant-col-sm-6 ant-col-md-5 ant-col-lg-3 ant-col-xl-3"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <img
            alt="fallback"
            src="./assets/images/footwear.jpg"
            style={
              Object {
                "height": "55px",
                "width": "55px",
              }
            }
          />
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <button
            className="ant-btn ant-btn-link ant-dropdown-trigger ant-dropdown-link category"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            <span>
               
            </span>
            <span
              aria-label="down"
              className="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="ant-divider ant-divider-vertical"
      role="separator"
      style={
        Object {
          "borderColor": "#baafaf",
          "borderWidth": "2px",
          "height": "50px",
          "margin": "2px",
        }
      }
    />
    <div
      className="ant-col gutter-row  ant-col-xs-16 ant-col-sm-6 ant-col-md-5 ant-col-lg-3 ant-col-xl-3"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        className="ant-space ant-space-horizontal ant-space-align-center"
        style={Object {}}
      >
        <div
          className="ant-space-item"
          style={
            Object {
              "marginRight": 8,
            }
          }
        >
          <img
            alt="fallback"
            src="./assets/images/bags.png"
            style={
              Object {
                "height": "55px",
                "width": "55px",
              }
            }
          />
        </div>
        <div
          className="ant-space-item"
          style={Object {}}
        >
          <button
            className="ant-btn ant-btn-link ant-dropdown-trigger ant-dropdown-link category"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            type="button"
          >
            <span>
               
            </span>
            <span
              aria-label="down"
              className="anticon anticon-down"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="down"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
